### PR TITLE
Feature/memory profiling

### PR DESCRIPTION
**Pull request by oscarvik**
memory profiling didn't give much valuable info, but did remove lots of unnecessary variables

### CODE REVIEW

File: trainer.py [hunk 1]
```diff
@@ -91,24 +76,21 @@ def train_transductive(
...
 
-    loss_ = model.lp_loss(pred, y)
-    loss = loss_ if loss is None else loss + loss_
+    loss = model.lp_loss(pred, y)
     lp_auc, lp_ap = model.lp_test(pred, y)
+
     if ((mode == "test")) and cal_mrr_score:
         model.lp_log_ranks(logits_lp, pei, pet, data.edge_index, data.edge_type)
-
-    total_loss += loss * num_graphs
```
Comment (oddaspa) [inline trainer.py:105]: The total_loss variable should still be there right?

Comment (oscarvik) [reply trainer.py:105]: We removed the loop "for data in loader"

Total_loss is always the same as loss

**Review by oddaspa — APPROVED** [review]
nice work :)
